test/node-http-test.js

```diff
@@ -94,10 +96,133 @@ describe('Node HTTP', function() {
+        assert.equal(0, balance.lockedBalance);
+        assert.equal(0, balance.lockedCoins.length);
+      }
+
+      {
+        // Fund address & verify balance.
+        const nextInterval = treeInterval;
+        const addedReward = nextInterval * consensus.BASE_REWARD;
+        const oBalance = await nclient.get(`/balance/${cbAddress}`);
+        await mineBlocks(nextInterval, cbAddress);
+        const nBalance = await nclient.get(`/balance/${cbAddress}`);
+
+        assert.equal(oBalance.height + nextInterval, nBalance.height);
+        assert.equal(oBalance.balance + addedReward, nBalance.balance);
+        assert.equal(oBalance.coins.length + nextInterval, nBalance.coins.length);
+        assert.equal(oBalance.lockedBalance, nBalance.lockedBalance);
+        assert.equal(oBalance.lockedCoins.length, nBalance.lockedCoins.length);
+      }
+
+      const name = await nclient.execute('grindname', [5]);
+      const rawName = Buffer.from(name, 'ascii');
+      const nameHash = rules.hashName(rawName);
+
+      {
+        // Create open.
+        const coins = await nclient.getCoinsByAddresses([cbAddress]);
+        coins.sort((a, b) => a.height - b.height);
+        const mtx = new MTX();
+        mtx.addCoin(Coin.fromJSON(coins[0]));
+        mtx.addOutput(new Output({
+          address: cbAddress,
+          value: 0,
+          covenant: {
+            type: types.OPEN,
+            items: [nameHash, Buffer.alloc(4), rawName]
+          }
+        }));
+        mtx.addOutput(new Output({
+          address: cbAddress,
+          value: 1999986000,
+          covenant: {
+            type: types.NONE,
+            items: []
+          }
+        }));
+        mtx.sign(keyring);
+        assert(mtx.verify());
+
+        // Mine to next interval & verify balance.
+        const nextInterval = treeInterval + 1;
+        const addedReward = nextInterval * consensus.BASE_REWARD;
+        const oBalance = await nclient.get(`/balance/${cbAddress}`);
+        await node.sendTX(mtx.toTX());
+        await mineBlocks(nextInterval, cbAddress);
+        const nBalance = await nclient.get(`/balance/${cbAddress}`);
+
+        assert.equal(oBalance.height + nextInterval, nBalance.height);
+        assert.equal(oBalance.balance + addedReward, nBalance.balance);
+        assert.equal(oBalance.coins.length + nextInterval, nBalance.coins.length);
+        assert.equal(oBalance.lockedBalance, nBalance.lockedBalance);
+        assert.equal(oBalance.lockedCoins.length + 1, nBalance.lockedCoins.length);
+      }
+
+      {
+        // Create bid.
+        const coins = await nclient.getCoinsByAddresses([cbAddress]);
+        coins.sort((a, b) => a.height - b.height);
+        const covenants = coins.filter((coin) => {
+          return coin.covenant.type === types.OPEN;
+        });
+        const mtx = new MTX();
+        const ns = await mtx.view.getNameState(node.chain.db, nameHash);
+        const height = Buffer.from([ns.height, 0x00, 0x00, 0x00]);
```

Comment:
This looks a little confusing -- can you use `covenant.push32()` after `mtx.addOutput()` but before `sign`ing ?